core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -505,6 +506,7 @@ public List<DeterministicKey> getKeys(KeyPurpose purpose, int numberOfKeys) {
             // and calculate the full lookahead zone there, so network requests will always use the right amount.
             List<DeterministicKey> lookahead = maybeLookAhead(parentKey, index, 0, 0);
             basicKeyChain.importKeys(lookahead);
+            lookahead.forEach(hierarchy::putKey);
```

Comment:
Why hasn't this been here before? Is this probably a (hidden) bugfix?

Reply:
It was done inside `maybeLookAhead()` -- you can see elsewhere in this PR that I removed it.

Reply:
It was done inside `maybeLookAhead()` -- you can see elsewhere in this PR that I removed it.

Reply:
Note that I also updated the comments on `maybeLookAhead()` to indicate that it is now side effect free.  (This was the major side effect -- adding keys to the hierarchy)